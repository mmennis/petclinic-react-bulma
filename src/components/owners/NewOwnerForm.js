import React from 'react'
import { Modal, Form, Button, Content, Heading } from 'react-bulma-components'
import PropTypes from 'prop-types'


const fieldStyle = {
}

const labelStyle = {
}

const errorStyle = {

}
const resetTouched = {
    first_name: false,
    last_name: false,
    address: false,
    city: false,
    state: false,
    telephone: false
}

export default class NewOwnerForm extends React.Component {

    constructor(props) {
        super(props)

        this.state = {
            show: false,
            newOwner: {},
            errors: {},
            touched: {...resetTouched},
            incomplete: true
        }

        this.open = this.open.bind(this)
        this.close = this.close.bind(this)
    }

    open = () => this.setState({ show: true })
    close = () => this.setState({ show: false, newVet: {}, errors: {}, touched: {...resetTouched}, incomplete: true })

    render() {
        return (
            <div>
                <Button onClick={this.open} size="small" data-testid="modal-open" color="primary">
                    New Owner
                </Button>
                <form data-testid="form">
                    <Modal show={this.state.show} onClose={this.close} {...this.props.modal}>
                        <Modal.Card>
                            
                        </Modal.Card>
                    </Modal>
                </form>
            </div>
        )
    }
}

NewOwnerForm.propTypes = {
    modal: PropTypes.object,
    handleNewOwner: PropTypes.func.isRequired
}