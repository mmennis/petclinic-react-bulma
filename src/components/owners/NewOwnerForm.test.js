import React from 'react'
import { create } from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'
import NewOwnerForm from './NewOwnerForm'

const newTestOwner = {
    first_name: 'first',
    last_name: 'last',
    address: '1234 Main St',
    city: 'Cupertino',
    state: 'CA',
    telephone: '(408) 555-1212'
}

describe('NewOwnerForm', () => {

    let handleNewOwnerCallback

    beforeEach(() => {
        handleNewOwnerCallback = jest.fn()
    })

    afterEach(() => {
        jest.clearAllMocks()
    })

    it('should render as per the snapshot', () => {
        const newOwnerForm = create(<NewOwnerForm handleNewOwner={handleNewOwnerCallback} />)
        expect(newOwnerForm.toJSON()).toMatchSnapshot()
    })

})