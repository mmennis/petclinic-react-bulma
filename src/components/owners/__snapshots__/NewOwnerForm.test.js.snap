// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOwnerForm should render as per the snapshot 1`] = `
<div>
  <button
    className="is-primary is-small button"
    data-testid="modal-open"
    disabled={false}
    onClick={[Function]}
    tabIndex={0}
  >
    New Owner
  </button>
  <form
    data-testid="form"
  />
</div>
`;
